src/map/addLayer.js

```diff
@@ -90,7 +90,7 @@ export default function (layer, mapConfig, dispatch) {
       layout: {},
       paint: {
         'circle-color':
-          layer.categories.color instanceof Array && !layer.paint
+          layer.categories && layer.categories.color instanceof Array && !layer.paint
```

Comment:
Seems like the only thing this branch brings into staging is checking that `layer.categories` exists before referencing `layer.categories.color`.